services/blockchain-connector/shared/sdk/cache.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const moment = require('moment');
+const BluebirdPromise = require('bluebird');
+
+const { getNodeInfo } = require('./endpoints_1');
+
+const { getTableInstance } = require('../database/better-sqlite3');
+const cacheBlockSchema = require('../database/schema/blocks');
+const cacheTrxIDToBlockIDSchema = require('../database/schema/transactions');
+
+const config = require('../../config');
+
+const getBlocksCache = () => getTableInstance(
+	cacheBlockSchema.tableName,
+	cacheBlockSchema,
+);
+
+const getTrxIDtoBlockIDCache = () => getTableInstance(
+	cacheTrxIDToBlockIDSchema.tableName,
+	cacheTrxIDToBlockIDSchema,
+);
+
+const blocksToCache = [];
+
+const cacheBlocksIfEnabled = async (blocks) => {
```

Comment:
Please wrap the entire logic within a try..catch block, so that the caller functions do not fail when `getNodeInfo` errors.